civis/tests/test_io.py

```diff
@@ -144,42 +149,353 @@ def test_large_file_to_civis(self, *mocks):
+        )
+        m_run_cleaning.return_value = [mock.sentinel.cleaning_future]
 
-    @mock.patch(api_import_str, return_value=civis_api_spec)
-    def test_civis_file_to_table(self, *mocks):
+        with mock.patch.object(
+                civis.io._tables, 'run_job',
+                spec_set=True) as m_run_job:
+
+            m_run_job.return_value = mock.sentinel.import_future
+
+            result = civis.io.civis_file_to_table(
+                mock_file_id, database, table,
+                existing_table_rows='truncate',
+                client=self.mock_client
+            )
+
+            assert result is mock.sentinel.import_future
+            m_run_job.assert_called_once_with(mock_import_id,
+                                              client=self.mock_client)
+
+        m_run_cleaning.assert_called_once_with(
+            [mock_file_id], self.mock_client, False, True
+        )
+        m_process_cleaning_results.assert_called_once_with(
+            [mock.sentinel.cleaning_future],
+            self.mock_client,
+            None,
+            False,
+            None
+        )
+
+        expected_name = 'CSV import to scratch.api_client_test_fixture'
+        expected_kwargs = {
+            'name': expected_name,
+            'max_errors': None,
+            'existing_table_rows': 'truncate',
+            'column_delimiter': 'comma',
+            'compression': 'gzip',
+            'escaped': False,
+            'execution': 'immediate',
+            'table_columns': None,
+            'redshift_destination_options': {
+                'diststyle': None, 'distkey': None,
+                'sortkeys': [None, None]
+            }
+
+        }
+        self.mock_client.imports.post_files_csv.assert_called_once_with(
+            {'file_ids': [mock_cleaned_file_id]},
+            {
+                'schema': 'scratch',
+                'table': 'api_client_test_fixture',
+                'remote_host_id': 42,
+                'credential_id': 713,
+                'primary_keys': None,
+                'last_modified_keys': None
+            },
+            True,
+            **expected_kwargs
+        )
+
+    @pytest.mark.civis_file_to_table
+    @mock.patch('civis.io._tables._process_cleaning_results')
+    @mock.patch('civis.io._tables._run_cleaning')
+    def test_civis_file_to_table_table_doesnt_exist(
+            self,
+            m_run_cleaning,
+            m_process_cleaning_results,
+            _m_get_api_spec
+    ):
         table = "scratch.api_client_test_fixture"
         database = 'redshift-general'
-        result = civis.io.civis_file_to_table(self.file_id, database, table,
-                                              existing_table_rows='truncate',
-                                              polling_interval=POLL_INTERVAL)
-        result = result.result()  # block until done
+        mock_file_id = 1234
+        mock_cleaned_file_id = 1235
+        mock_import_id = 8675309
```

Comment:
Ha, glad to see someone else uses this as a mock number (although you might be copying it from my code in the client, I suppose)